boards/arm/arduino_portenta_h7/doc/index.rst

```diff
@@ -0,0 +1,152 @@
+More information about STM32H747XIH6 can be found here:
+
+- `STM32H747XI on www.st.com`_
+- `STM32H747xx reference manual`_
+- `STM32H747xx datasheet`_
+
+Supported Features
+==================
+
+The current Zephyr arduino_portenta_h7 board configuration supports the following hardware features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| RNG       | on-chip    | True Random number generator        |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| IPM       | on-chip    | virtual mailbox based on HSEM       |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+Resources sharing
+=================
+
+The dual core nature of STM32H747 SoC requires sharing HW resources between the
+two cores. This is done in 3 ways:
+
+- **Compilation**: Clock configuration is only accessible to M7 core. M4 core only
+  has access to bus clock activation and deactivation.
+- **Static pre-compilation assignment**: Peripherals such as a UART are assigned in
+  devicetree before compilation. The user must ensure peripherals are not assigned
+  to both cores at the same time.
+- **Run time protection**: Interrupt-controller and GPIO configurations could be
+  accessed by both cores at run time. Accesses are protected by a hardware semaphore
+  to avoid potential concurrent access issues.
+
+Programming and Debugging
+*************************
+
+Applications for the ``arduino_portenta_h7`` board should be built per core target,
+using either ``arduino_portenta_h7_m7`` or ``arduino_portenta_h7_m4`` as the target.
+See :ref:`build_an_application` for more information about application builds.
+
+
+Flashing
+========
+
+Flashing an application to STM32H747I M7 Core
+---------------------------------------------
+
+First, connect the Arduino Portenta H7 board to your host computer using
+the USB port to prepare it for flashing. Then build and flash your application.
```

Comment:
Please provide more detailed instructions.
As this is DFU this is not as direct as flashing using openocd/ST-Link.

Reply:
Should of course add instructions about double tapping before flashing. But other than that I'm just using west flash as always. Should I add instructions about installing dfu-utils or just mention that it's required?

Reply:
> Should I add instructions about installing dfu-utils or just mention that it's required?

Not required. As dfu is already supported by west, one should be able to find dfu instructions as part of west doc (at least I hope so).